src/d/actor/d_a_player_swim.inc

```diff
@@ -11,85 +11,613 @@
+                voiceStart(0x21);
+                mFrameCtrlUnder[0].mRate = 0.0f;
+                onNoResetFlg1(daPyFlg1_LETTER_READ_EYE_MOVE); // the `oris` instruction for this code suggests `onNoResetFlg1` needs to include a `<< 16` op but that breaks other functions
+            }
+            return FALSE;
+        }
+    }
 }
 
 /* 8013D530-8013D638       .text setSwimMoveAnime__9daPy_lk_cFQ29daPy_lk_c8daPy_ANM */
-void daPy_lk_c::setSwimMoveAnime(daPy_ANM) {
+void daPy_lk_c::setSwimMoveAnime(daPy_ANM swimMoveAnm) {
     /* Nonmatching */
+    //float startFrame = mFrameCtrlUnder[0].getFrame();
+    float endFrame = mFrameCtrlUnder[0].getEnd();
+    setSingleMoveAnime(swimMoveAnm, (std::fabsf(mVelocity) * ((l_HIO_swim_c0.m.field_0x54 - l_HIO_swim_c0.m.field_0x50))) / mMaxNormalSpeed + l_HIO_swim_c0.m.field_0x50 + (float)(getSwimTimerRate() * l_HIO_swim_c0.m.field_0x74), 0.0f, -1, l_HIO_swim_c0.m.field_0x58); // Missing a `* currentFrame` in here?
+    mFrameCtrlUnder[0].setFrame(endFrame * mFrameCtrlUnder[0].getEnd());
+    mAnmRatioUnder[0].getAnmTransform()->setFrame(mFrameCtrlUnder[0].getFrame());
 }
 
 /* 8013D638-8013D6B8       .text getSwimTimerRate__9daPy_lk_cFv */
-void daPy_lk_c::getSwimTimerRate() {
-    /* Nonmatching */
+f32 daPy_lk_c::getSwimTimerRate() {
+    f32 swimTimerRate;
+    if (dComIfGp_getItemTimeMax() && !dComIfGp_event_getMode()) { // fakematch? possibly the dComIfGp_getItemSwimTimerStatus inline mentioned in the debug maps
+        swimTimerRate = (1.0f - dComIfGp_getItemTimeCount() * 0.0011111111f);
+    }
+    else {
+        if (dComIfGp_getItemTimeCount() <= 0) {
+            swimTimerRate = 1.0f;
+        }
+        else {
+            swimTimerRate = 0.0f;
+        }
+    }
+    return swimTimerRate;
 }
 
 /* 8013D6B8-8013D814       .text setSwimTimerStartStop__9daPy_lk_cFv */
 void daPy_lk_c::setSwimTimerStartStop() {
-    /* Nonmatching */
+    float target = 0.0f;
+
+    if (mAcch.GetGroundH() <= m35D0 - 175.0f) {
+        dComIfGp_setItemTimeMax(1);
+
+        if (checkNoResetFlg0(daPyFlg0_UNK100)) {
+            float swimTimerRate = getSwimTimerRate();
+
+            if (swimTimerRate > 0.5f) {
+                target = cM_scos(cM_rad2s((1.0f - swimTimerRate) * 3.1415927f)) * -l_HIO_swim_c0.m.field_0x78;
+
+                if (mFrameCtrlUnder->checkPass(0.0f) && mCurProc != 0x35) {
+                    if (m3570 && !checkNoResetFlg0(daPyFlg0_UNK4000)) {
+                        voiceStart(0x22);
+                        m3570 = 0;
+                    }
+                    else {
+                        m3570 = 1;
+                    }
+                }
+            }
+        }
+    }
+    else {
+        dComIfGp_setItemTimeCount(900);
+        dComIfGp_setItemTimeMax(1);
+        dComIfGp_setItemTimeMax(0); //Fakematch? Seems nonsensical to set it to 1 then immediately to 0, also could be `dComIfGp_stopItemSwimTimer` and `dComIfGp_startItemSwimTimer` inlines mentioned in the debug maps
```

Comment:
Yeah something probably isn't right here, I suspect some of these inlines do more than one thing based on their byte sizes in the debug maps. It's fine like this for now though, as more code gets decompiled we can narrow down the possibilities